kapitan/cli.py

```diff
@@ -409,19 +430,46 @@ def secret_update_validate(args, ref_controller):
                 logger.warning("WARNING: parameters.kapitan.secrets.recipients is deprecated, " +
                     "please use parameters.kapitan.secrets.gpg.recipients")
                 recipients = inv['nodes'][target_name]['parameters']['kapitan']['secrets']['recipients']
-            for token_path in token_paths:
-                secret_obj = ref_controller.backends['gpg'][token_path]
+        except KeyError:
+            recipients = None
+
+        try:
+            key = inv['nodes'][target_name]['parameters']['kapitan']['secrets']['gkms']['key']
+        except KeyError:
+            key = None
+
+        for token_path in token_paths:
+            if token_path.startswith("?{gpg:"):
+                if not recipients:
+                    logger.debug("secret_update_validate: target: %s has no inventory gpg recipients, skipping %s", target_name, token_path)
+                    continue
+                secret_obj = ref_controller[token_path]
                 target_fingerprints = set(lookup_fingerprints(recipients))
                 secret_fingerprints = set(lookup_fingerprints(secret_obj.recipients))
                 if target_fingerprints != secret_fingerprints:
                     if args.validate_targets:
-                        logger.info("%s recipient mismatch", token_path)
+                        logger.info("%s recipients mismatch", token_path)
```

Comment:
You could maybe add which fingerprints are different with:
```
target_fingerprints .symmetric_difference(secret_fingerprints)
```

Reply:
good idea